core/Plugin/Visualization.php

```diff
@@ -461,10 +461,30 @@ private function postDataTableLoadedFromAPI()
         if ($this->dataTable instanceof DataTable) {
             $this->metadata = $this->dataTable->getAllTableMetadata();
 
-            if (isset($this->metadata[DataTable::ARCHIVED_DATE_METADATA_NAME])) {
-                $this->reportLastUpdatedMessage = $this->makePrettyArchivedOnText();
+
+        } else {
+            // if the dataTable is Map
+            if ($this->dataTable instanceof DataTable\Map) {
+                // load all the data
+                $dataTable = $this->dataTable->getDataTables();
+                // find the latest key
+                foreach ($dataTable as $item) {
+                    $metaData = $item->getAllTableMetadata();
+                    // if ts_created not exist exit loop
+                    if (!isset($metaData[DataTable::ARCHIVED_DATE_METADATA_NAME])) {
+                        break;
+                    }
+                    // if ts_created not record in the metadata or if current is more recent up the metadata
+                    if (!isset($this->metadata[DataTable::ARCHIVED_DATE_METADATA_NAME])
+                      || strtotime($metaData[DataTable::ARCHIVED_DATE_METADATA_NAME]) > strtotime($this->metadata[DataTable::ARCHIVED_DATE_METADATA_NAME])) {
+                        $this->metadata = $metaData;
+                    }
+               }
             }
         }
+        if (isset($this->metadata[DataTable::ARCHIVED_DATE_METADATA_NAME])) {
```

Comment:
I know it has been that way before, but wouldn't an `!empty` check make more sense here? If for any reason the metadata entry exist, but has no value it shouldn't be used